src/masks/InteractionMasks.tsx

```diff
@@ -140,53 +129,85 @@ export default function InteractionMasks<R>({
     };
   }
 
-  function getNextPosition(keyCode: number) {
-    switch (keyCode) {
-      case KeyCodes.ArrowUp:
-        return { ...selectedPosition, rowIdx: selectedPosition.rowIdx - 1 };
-      case KeyCodes.ArrowDown:
-        return { ...selectedPosition, rowIdx: selectedPosition.rowIdx + 1 };
-      case KeyCodes.ArrowLeft:
-        return { ...selectedPosition, idx: selectedPosition.idx - 1 };
-      case KeyCodes.ArrowRight:
-        return { ...selectedPosition, idx: selectedPosition.idx + 1 };
+  function getNextPosition(key: string, mode = cellNavigationMode, shiftKey = false) {
+    const { idx, rowIdx } = selectedPosition;
+    let nextPosition: Position;
+    switch (key) {
+      case 'ArrowUp':
+        nextPosition = { idx, rowIdx: rowIdx - 1 };
+        break;
+      case 'ArrowDown':
+        nextPosition = { idx, rowIdx: rowIdx + 1 };
+        break;
+      case 'ArrowLeft':
+        nextPosition = { idx: idx - 1, rowIdx };
+        break;
+      case 'ArrowRight':
+        nextPosition = { idx: idx + 1, rowIdx };
+        break;
+      case 'Tab':
+        nextPosition = shiftKey ? { idx: idx - 1, rowIdx } : { idx: idx + 1, rowIdx };
+        break;
       default:
-        return selectedPosition;
+        nextPosition = { idx, rowIdx };
+        break;
     }
-  }
 
-  function onKeyDown(e: React.KeyboardEvent<HTMLDivElement>): void {
-    if (isCtrlKeyHeldDown(e)) {
-      onPressKeyWithCtrl(e);
-    } else if (e.keyCode === KeyCodes.Escape) {
-      onPressEscape();
-    } else if (e.keyCode === KeyCodes.Tab) {
-      onPressTab(e);
-    } else if (isKeyboardNavigationEvent(e)) {
-      changeCellFromEvent(e);
-    } else if (isKeyPrintable(e.keyCode) || [KeyCodes.Backspace, KeyCodes.Delete, KeyCodes.Enter].includes(e.keyCode)) {
-      openEditor(e);
-    }
+    return getNextSelectedCellPosition({
+      columns,
+      rowsCount: rows.length,
+      cellNavigationMode: mode,
+      nextPosition
+    });
   }
 
-  function openEditor(event: React.KeyboardEvent<HTMLDivElement>): void {
-    if (!isEditorEnabled && isCellEditable(selectedPosition)) {
-      setFirstEditorKeyPress(event.key);
-      setIsEditorEnabled(true);
+  function onKeyDown(event: React.KeyboardEvent<HTMLDivElement>): void {
+    const column = columns[selectedPosition.idx];
+    const row = rows[selectedPosition.rowIdx];
+    const isActivedByUser = (column.unsafe_onCellInput ?? legacyCellInput)(event, row) === true;
```

Comment:
`Actived` isn't a word.
```suggestion
    const isActivatedByUser = (column.unsafe_onCellInput ?? legacyCellInput)(event, row) === true;
```
`Triggered` is also good.